src/ngraph/runtime/dynamic/dynamic_backend.cpp

```diff
@@ -117,130 +117,158 @@ bool runtime::dynamic::DynamicExecutable::call(
     // (1) all shapes;
     // (2) all values of shape-relevant input tensors.
 
-    NGRAPH_CHECK(m_wrapped_function->get_parameters().size() == inputs.size());
+    auto lru = std::make_shared<runtime::LRUCache>(4);
+    std::vector<size_t> merged_input_shapes;
+    std::ostringstream key;
 
-    std::vector<std::shared_ptr<runtime::Tensor>> wrapped_inputs;
-    std::vector<element::Type> arg_element_types;
-    std::vector<PartialShape> arg_shapes;
+    for (auto& input : inputs)
```

Comment:
With variable rank, this will do the wrong thing: [2], [3, 4] and [2, 3], [4] will get the same key.
You can directly use a vector of shapes as a key and skip the conversion to a string.